handler/src/main/java/io/netty/handler/ssl/Java9SslUtils.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.ssl;
+
+import io.netty.util.internal.EmptyArrays;
+import io.netty.util.internal.PlatformDependent;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLParameters;
+import java.lang.reflect.Method;
+import java.security.AccessController;
+import java.security.PrivilegedExceptionAction;
+import java.util.List;
+
+final class Java9SslUtils {
+    private static final InternalLogger log = InternalLoggerFactory.getInstance(Java9SslUtils.class);
+    private static final boolean supportsAlpn;
+    private static final Method setApplicationProtocols;
+    private static final Method getApplicationProtocol;
+
+    static {
+        boolean support = false;
+        Method setter = null;
+        Method getter = null;
+
+        if (PlatformDependent.javaVersion() >= 9) {
+            try {
+                getter = AccessController.doPrivileged(new PrivilegedExceptionAction<Method>() {
+                    @Override
+                    public Method run() throws Exception {
+                        return SSLEngine.class.getMethod("getApplicationProtocol");
+                    }
+                });
+                setter = AccessController.doPrivileged(new PrivilegedExceptionAction<Method>() {
+                    @Override
+                    public Method run() throws Exception {
+                        return SSLParameters.class.getMethod("setApplicationProtocols", String[].class);
```

Comment:
If I use Netty in my project but not Netty's alpn and this is invoked, will it cause problems because setApplicationProtocols is invoked with an empty array?